### DISCUSSION TITLE
How to put the file-like object into an s3 bucket

### DISCUSSION THREAD

**Discussion by closet6** [fastapi/fastapi#7680] [Questions] [answered]
### First check

* [ x] I used the GitHub search to find a similar issue and didn't find it.
* [x ] I searched the FastAPI documentation, with the integrated search.
* [ x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Using file: 

`file: UploadFile = File(...) `

I try to put the file-like [object into an s3 bucket](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Bucket.put_object):

`s3_response = s3.Bucket(bucket_name).put_object(Key=file.filename, Body=file)`

i get the error: 
```

start_position = fileobj.tell()
AttributeError: 'UploadFile' object has no attribute 'tell'
```

I'm assuming i need to either implement tell(), or convert to a bytes

Would there be other options?

And if so, how you recommend into converting UploadFile into a bytes efficiently?

**Comment by merowinger92** [comment, marked as answer]
`UploadedFile` has a member `file` which might help. So instead of using `s3_response = s3.Bucket(bucket_name).put_object(Key=file.filename, Body=file)` it might help using `s3_response = s3.Bucket(bucket_name).put_object(Key=file.filename, Body=file.file)`

**Comment by closet6** [comment]
that worked, thank you

**Comment by tiangolo** [comment]
Thanks for the help here @merowinger92 ! :bowing_man: :cake: 

Thanks for reporting back and closing the issue @joeraygme :+1:

**Comment by jcf-dev** [comment]
how can you this asynchronously or what if file is big? How we can do this chunk by chunk per se.

**Comment by ycd** [comment]
@joweenflores check out the asyncio supported version: https://github.com/aio-libs/aiobotocore
